src/index.js

```diff
@@ -204,381 +253,378 @@ class Masq {
-          registering = true
-          this._requestUserAppRegister(key, peer)
-          break
-
-        case 'masqAccessGranted':
-          registering = false
-
-          userId = json.userAppDbId
-          dataEncryptionKey = json.userAppDEK
-          username = json.username
-          nonce = json.userAppNonce
-          profileImage = json.profileImage
-
-          const buffKey = Buffer.from(json.key, 'hex')
-          db = common.utils.createPromisifiedHyperDB(userId, buffKey)
-          await common.utils.dbReady(db)
-
-          waitingForWriteAccess = true
-          this._requestWriteAccess(key, peer, db.local.key)
-          break
-
-        case 'masqAccessRefused':
-          registering = false
-          handleError(new MasqError(MasqError.MASQ_ACCESS_REFUSED_BY_USER))
-          return
-
-        case 'writeAccessGranted':
-          waitingForWriteAccess = false
-
-          // Store the session info
-          this._storeSessionInfo(stayConnected, userId, dataEncryptionKey, username, profileImage, nonce)
-          this.userId = userId
-          this.nonce = nonce
-          this.dataEncryptionKey = await common.crypto.importKey(Buffer.from(dataEncryptionKey, 'hex'))
-
-          this.userAppDb = db
-          this._startReplication()
-
-          sw.close()
-          break
-      }
+        } catch (err) {
+          await this._resetLogin('could not decrypt')
+          reject(err)
+        }
+        resolve()
+      })
+    })
+
+    await pr
+  }
+
+  async _dbCreation () {
+    debug('[masq._dbCreation]')
+    this.setState('userAppDbCreated')
+
+    await this._openDb()
+
+    const msg = {
+      msg: 'connectionEstablished'
     }
+    const encryptedMsg = await common.crypto.encrypt(this.loginKey, msg, 'base64')
 
-    this._initSwarmWithDataHandler(channel, handleData).then(
-      () => {
-        this._logIntoMasqDone = true
-        if (this._onLogIntoMasqDone) this._onLogIntoMasqDone()
-      },
-      (e) => {
-        this._logIntoMasqErr = e
-        this._logIntoMasqDone = true
-        if (this._onLogIntoMasqDone) this._onLogIntoMasqDone()
-      }
-    )
+    this.loginPeer.send(JSON.stringify(encryptedMsg))
 
-    return new Promise((resolve, reject) => {
-      if (this._logIntoMasqDone) {
-        this._logIntoMasqDone = false
-        if (this._logIntoMasqErr) {
-          return reject(this._logIntoMasqErr)
-        } else {
-          this.eventTarget.dispatchEvent(new Event('logged_in'))
-          return resolve()
-        }
-      }
-      this._onLogIntoMasqDone = () => {
-        this._onLogIntoMasqDone = undefined
-        this._logIntoMasqDone = false
-        if (this._logIntoMasqErr) {
-          return reject(this._logIntoMasqErr)
-        } else {
-          this.eventTarget.dispatchEvent(new Event('logged_in'))
-          return resolve()
-        }
-      }
-    })
+    await this._awaitReadyMsg()
+  }
+
+  async _awaitReadyMsg () {
```

Comment:
I don't understand the purpose of this function, from the function's name and the comment we can expect for a 'ready' message but this function set a state and start the replication, which is not expected from the name

Reply:
We were supposed to await a 'ready' message sent from Masq app before finishing the login procedure and starting the replication but it is not sent by masq-app so we don't await the msg.